tests/services/intelligence/test_preference_storage.py

```
@@ -0,0 +1,406 @@
+# Protocol-contract tests (parametrized over both implementations)
+# ---------------------------------------------------------------------------
+
+
+@pytest.fixture(params=["in_memory", "sqlite"])
+def storage(request: pytest.FixtureRequest, tmp_path: Path) -> PreferenceStorage:
+    """Yield each storage implementation under test."""
+    if request.param == "in_memory":
+        return InMemoryPreferenceStorage()
+    if request.param == "sqlite":
+        return SqlitePreferenceStorage(tmp_path / "preferences.db")
+    raise ValueError(f"Unknown storage param: {request.param}")
```

Ruff (0.15.11)

[warning] Avoid specifying long messages outside the exception class

(TRY003)